FakeRate/app_region/ScanChain.C

```diff
@@ -512,147 +523,170 @@ int ScanChain( TChain* chain, TString fakeratefile, TString option = "", TString
-		if (abs(ss.lep1_id())==11) {
-		  pTrelvsIso_histo_el->Fill( std::min(ss.lep1_iso(),float(0.99)), std::min(lep1_ptrel_v1,float(29.9)) );
-		  pTrelvsMiniIso_histo_el->Fill( std::min(ss.lep1_miniIso(),float(0.99)), std::min(lep1_ptrel_v1,float(29.9)) );
+	    if( ss::lep1_id()*ss::lep2_id() > 0 ) {
+	      if (ss::lep1_motherID()<=0 && /*ss::lep1_iso()>0.1 &&*/ fabs(ss::lep1_ip3d()/ss::lep1_ip3d_err())<4. && ss::lep2_motherID()==1) {
+		if (abs(ss::lep1_id())==11) {
+		  pTrelvsIso_histo_el->Fill( std::min(ss::lep1_iso(),float(0.99)), std::min(lep1_ptrel_v1,float(29.9)) );
+		  pTrelvsMiniIso_histo_el->Fill( std::min(ss::lep1_miniIso(),float(0.99)), std::min(lep1_ptrel_v1,float(29.9)) );
 		  hists[getHist("pTrel_histo_el")]->Fill(std::min(lep1_ptrel_v1,float(29.9)) );
 		} else {
-		  pTrelvsIso_histo_mu->Fill( std::min(ss.lep1_iso(),float(0.99)), std::min(lep1_ptrel_v1,float(29.9)) );
-		  pTrelvsMiniIso_histo_mu->Fill( std::min(ss.lep1_miniIso(),float(0.99)), std::min(lep1_ptrel_v1,float(29.9)) );
+		  pTrelvsIso_histo_mu->Fill( std::min(ss::lep1_iso(),float(0.99)), std::min(lep1_ptrel_v1,float(29.9)) );
+		  pTrelvsMiniIso_histo_mu->Fill( std::min(ss::lep1_miniIso(),float(0.99)), std::min(lep1_ptrel_v1,float(29.9)) );
 		  hists[getHist("pTrel_histo_mu")]->Fill(std::min(lep1_ptrel_v1,float(29.9)) );
 		}
 	      }
-	      if (ss.lep2_motherID()<=0 && /*ss.lep2_iso()>0.1 &&*/ fabs(ss.lep2_ip3d()/ss.lep2_ip3d_err())<4. && ss.lep1_motherID()==1) {
-		if (abs(ss.lep2_id())==11) {
-		  pTrelvsIso_histo_el->Fill( std::min(ss.lep2_iso(),float(0.99)), std::min(lep2_ptrel_v1,float(29.9)) );
-		  pTrelvsMiniIso_histo_el->Fill( std::min(ss.lep2_miniIso(),float(0.99)), std::min(lep2_ptrel_v1,float(29.9)) );
+	      if (ss::lep2_motherID()<=0 && /*ss::lep2_iso()>0.1 &&*/ fabs(ss::lep2_ip3d()/ss::lep2_ip3d_err())<4. && ss::lep1_motherID()==1) {
+		if (abs(ss::lep2_id())==11) {
+		  pTrelvsIso_histo_el->Fill( std::min(ss::lep2_iso(),float(0.99)), std::min(lep2_ptrel_v1,float(29.9)) );
+		  pTrelvsMiniIso_histo_el->Fill( std::min(ss::lep2_miniIso(),float(0.99)), std::min(lep2_ptrel_v1,float(29.9)) );
 		  hists[getHist("pTrel_histo_el")]->Fill(std::min(lep2_ptrel_v1,float(29.9)) );
 		} else {
-		  pTrelvsIso_histo_mu->Fill( std::min(ss.lep2_iso(),float(0.99)), std::min(lep2_ptrel_v1,float(29.9)) );
-		  pTrelvsMiniIso_histo_mu->Fill( std::min(ss.lep2_miniIso(),float(0.99)), std::min(lep2_ptrel_v1,float(29.9)) );
+		  pTrelvsIso_histo_mu->Fill( std::min(ss::lep2_iso(),float(0.99)), std::min(lep2_ptrel_v1,float(29.9)) );
+		  pTrelvsMiniIso_histo_mu->Fill( std::min(ss::lep2_miniIso(),float(0.99)), std::min(lep2_ptrel_v1,float(29.9)) );
 		  hists[getHist("pTrel_histo_mu")]->Fill(std::min(lep2_ptrel_v1,float(29.9)) );
 		}
 	      }
 	    }
 	  }
 
-  if (ss.hyp_class() == 3){
+
+  if (ss::hyp_class() == 3 && lep1_pT >= 20 && lep2_pT >= 20){
```

Comment:
Hi Giuseppe,

About the 20/20 - I agree, this whole change is unnecessary, will revert
it..

About the class files -- this is just allowing this directory to use the
same class files as everywhere else.

Maybe we should move this discussion to private e-mail, I think these
comments are going to the whole SNT list !!

-Alex

On Fri, Jun 19, 2015 at 12:58 AM, cerati <notifications@github.com> wrote:

> In FakeRate/app_region/ScanChain.C
> <https://github.com/cmstas/SSAnalysis/pull/3#discussion_r32808217>:
>
> >  		  hists[getHist("pTrel_histo_mu")]->Fill(std::min(lep2_ptrel_v1,float(29.9)) );
> >  		}
> >  	      }
> >  	    }
> >  	  }
> >
> > -  if (ss.hyp_class() == 3){
> > +
> > +  if (ss::hyp_class() == 3 && lep1_pT >= 20 && lep2_pT >= 20){
>
> why >20 for both legs? it won't work for LL and HL
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cmstas/SSAnalysis/pull/3/files#r32808217>.
>